tests/test_file_upload.ts

```diff
@@ -0,0 +1,346 @@
+import FileUpload from '@/components/file_upload.vue';
+import Modal from '@/components/modal.vue';
+import { config, mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+
+import { patch_component_method } from './mocking';
+
+beforeAll(() => {
+    config.logModifiedComponents = false;
+});
+
+interface HTMLInputEvent extends Event {
+    target: HTMLInputElement & EventTarget;
+}
+
+class MockFileList implements FileList {
+    mock_files: File[];
+
+    constructor(files: File[]) {
+        this.mock_files = files;
+    }
+
+    item(index: number) {
+        return this.mock_files[index];
+    }
+
+    get length() {
+        return this.mock_files.length;
+    }
+
+    [Symbol.iterator]() {
+        return this.mock_files[Symbol.iterator]();
+    }
+
+    [index: number]: File;
+}
+
+describe('File Upload tests not involving nested modal component', () => {
+    let wrapper: Wrapper<FileUpload>;
+    let file_upload_component: FileUpload;
+    let empty_files_present_modal: Modal;
+    let rows1: string[];
+    let rows2: string[];
+    let rows3: string[];
+    let rows4: string[];
+    let fake_file_1: File;
```

Comment:
These files aren't actually "fake" (they're non-mock file objects), so remove the "fake" from the name.